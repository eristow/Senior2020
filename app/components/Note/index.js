/**
 *
 * Note
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

import NoteWhite from './NoteWhite';
import NoteBlack from './NoteBlack';
// import { FormattedMessage } from 'react-intl';
// import messages from './messages';

// TODO: how to get FormattedMessage from props.name
// TODO: how to get audio from props.name
function Note(props) {
  return (
    <div>
      {props.color === 'white' ? (
        <NoteWhite
          style={props.style}
          onClick={() => props.onClick(props.note)}
        >
          {props.name}
          {/* <FormattedMessage {...props.name} /> */}
        </NoteWhite>
      ) : (
        <NoteBlack
          style={props.style}
          onClick={() => props.onClick(props.note)}
        >
          {props.name}
          {/* <FormattedMessage {...props.name} /> */}
        </NoteBlack>
      )}
    </div>
  );
}

Note.propTypes = {
  style: PropTypes.object,
  name: PropTypes.string,
  note: PropTypes.string,
  color: PropTypes.string,
  onClick: PropTypes.func,
};

export default Note;
