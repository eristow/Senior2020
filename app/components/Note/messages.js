/*
 * Note Messages
 *
 * This contains all the text for the Note component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Note';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the Note component!',
  },
});
