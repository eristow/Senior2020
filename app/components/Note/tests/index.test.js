/**
 *
 * Tests for Note
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render, fireEvent } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Note from '../index';

describe('<Note />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Note />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Note />);
    expect(firstChild).toMatchSnapshot();
  });

  it('Should handle click on NoteWhite', () => {
    const onClickMock = jest.fn();
    const name = 'White Note';
    const props = {
      name,
      color: 'white',
      onClick: onClickMock,
    };

    const { container } = render(<Note {...props} />);

    fireEvent.click(container.querySelector('button'));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('Should handle click on NoteBlack', () => {
    const onClickMock = jest.fn();
    const name = 'Black Note';
    const props = {
      name,
      color: 'black',
      onClick: onClickMock,
    };

    const { container } = render(<Note {...props} />);

    fireEvent.click(container.querySelector('button'));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
