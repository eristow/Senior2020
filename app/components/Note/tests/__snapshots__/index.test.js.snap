// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> Should render and match the snapshot 1`] = `
<div>
  <button
    class="NoteBlack-jr31tw-0 fQVXRh"
  />
</div>
`;
